### PR TITLE
Azure does not display the tabs

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: -v cfme/tests/cloud/test_instance_power_control.py --long-running --use-provider azure -k "not cancel" }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit b7b69ea7b107c533da2e64959898366bdfe83ed4:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by jteehan** [comment]
I like this a lot better.  I agree that providers should not be worrying about the underlying intricacies of html changes.  I had been rebasing every day hoping my azure issue would just go away.  :)

**Comment by mfalesni** [comment]
@jteehan So, I cherrypicked your commit without the "gut" changes and it seems to work. I will comment on some lines to show the differences between your and this implementation.

File: cfme/cloud/instance.py [hunk 1]
```diff
@@ -367,7 +369,12 @@ class AzureInstance(Instance):
     START = "Start"
     POWER_ON = START  # For compatibility with the infra objects.
     STOP = "Stop"
-    TERMINATE = "Terminate"
+    SUSPEND = "Suspend"
+    DELETE = "Delete"
+    TERMINATE = deferred_verpick({
+        version.LOWEST: 'Terminate',
+        '5.6.1': 'Delete',
+    })
```
Comment (mfalesni) [inline cfme/cloud/instance.py]: I used a deferred verpick here, which only accesses the SSH of the appliance after accessed for the first time.

File: cfme/cloud/provider/azure.py [hunk 2]
```diff
@@ -27,7 +28,14 @@ def _form_mapping(self, create=None, **kwargs):
...
         credentials = cls.process_credential_yaml_key(credentials_key)
-        return cls(name=prov_config['name'],
+        # HACK: stray domain entry in credentials, so ensure it is not there
+        credentials.domain = None
+        region = prov_config.get('region', None)
+        if region is not None and isinstance(region, dict):
+            region = pick(region)
+        return cls(
+            name=prov_config['name'],
+            region=region,
```
Comment (mfalesni) [inline cfme/cloud/provider/azure.py]: @jteehan Here I did not bother with verpicking in the code, but rather keeping the values in the yaml. We can't verpick all the values in the yaml, but with region we know there can be differences. In the yaml, you don't have to use the dictionary, as you see here, it is optional. And it also passes the region so it then subsequently gets inserted in the UI.

**Comment by dajoRH** [comment]
Lint report for commit c8fd9976e0e3309cbc37bbdb45a5da46b458cff7:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/common/provider.py [hunk 3]
```diff
@@ -71,7 +71,7 @@ def form(self):
                 ('google_service_account', Input('service_account')),
             ]
             tab_fields = {
-                "Default": [
+                ("Default", ('default_when_no_tabs', )): [
```
Comment (mfalesni) [inline cfme/common/provider.py]: @jteehan This is the architectural thing I was talking about. This sets a flag on the tab that if no tabs are there, then if accessing a field that is under this tab do nothing. And if accessing a different tab, cry.

File: cfme/tests/cloud/test_instance_power_control.py [hunk 4]
```diff
@@ -26,12 +27,15 @@ def testing_instance(request, setup_provider, provider):
...
         request.addfinalizer(instance.delete_from_provider)
+        provider.refresh_provider_relationships()
     elif instance.provider.type == "ec2" and \
             provider.mgmt.is_vm_state(instance.name, provider.mgmt.states['deleted']):
         provider.mgmt.set_name(
             instance.name, 'test_terminated_{}'.format(fauxfactory.gen_alphanumeric(8)))
         instance.create_on_provider(allow_skip="default", find_in_cfme=True)
         request.addfinalizer(instance.delete_from_provider)
+        provider.refresh_provider_relationships()
+    instance.wait_to_appear()
```
Comment (mfalesni) [inline cfme/tests/cloud/test_instance_power_control.py]: @jteehan This file is almost all yours except these three lines I added.

**Comment by dajoRH** [comment]
Lint report for commit 3a1ce3fae596ef1630169925bc61d09534f291a1:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by mfalesni** [comment]
Closing in favour of #3219 which contains the changes from this PR.

**Pull request closed without merging**